File: app/src/main/java/mattframpton/endclothing_mattframpton/di/application/EndClothingApplication.java
package mattframpton.endclothing_mattframpton.di.application;

import android.app.Application;

import mattframpton.endclothing_mattframpton.di.components.DaggerEndClothingApplicationComponent;
import mattframpton.endclothing_mattframpton.di.components.EndClothingApplicationComponent;
import mattframpton.endclothing_mattframpton.network.EndClothingService;

public class EndClothingApplication extends Application {
	private EndClothingService client;

	@Override
	public void onCreate() {
		injectDependencies();
		super.onCreate();
	}

	private void injectDependencies() {
		EndClothingApplicationComponent component = DaggerEndClothingApplicationComponent.builder().build();
		client = component.getEndClothingService();
	}

	public EndClothingService getEndClothingService(){
		return client;
	}
}

File: app/src/main/java/mattframpton/endclothing_mattframpton/di/application/EndClothingApplication.kt
package mattframpton.endclothing_mattframpton.di.application

import android.app.Application
import mattframpton.endclothing_mattframpton.di.components.DaggerEndClothingApplicationComponent

import mattframpton.endclothing_mattframpton.network.EndClothingService

class EndClothingApplication : Application() {
    var endClothingService: EndClothingService? = null
        private set

    override fun onCreate() {
        injectDependencies()
        super.onCreate()
    }

    private fun injectDependencies() {
        val component = DaggerEndClothingApplicationComponent.builder().build()
        endClothingService = component.endClothingService
    }
}
